dev/test/pool.ts

```diff
@@ -327,4 +327,25 @@ describe('Client pool', () => {
         expect(err).to.equal('The client has already been terminated');
       });
   });
+
+  it('waits for existing operations to complete before releasing clients', async () => {
+    const clientPool = new ClientPool<{}>(1, 0, () => {
+      return {};
+    });
+    const deferred = new Deferred<void>();
+
+    const op = clientPool.run(REQUEST_TAG, async () => {
+      await deferred.promise;
+      return Promise.resolve('success');
+    });
+    const terminate = clientPool
+      .terminate()
+      .then(() => Promise.resolve('success'));
+    deferred.resolve();
+
+    // Wait for the terminate promise to resolve after run() completes.
+    await terminate;
+    expect(op).to.become('success');
```

Comment:
If you keep this, I think you need to call `await` since `become` is a PromisedAssertion.